common/src/main/java/com/wynntils/features/user/UpdateReminderFeature.java

```diff
@@ -34,33 +30,22 @@ public void onWorldStateChange(WorldStateEvent event) {
 
         firstJoin = false;
 
-        Request versionFetch = new RequestBuilder(CI_LINK, "latest_build")
-                .handleJsonObject((jsonObject) -> {
-                    int buildNumber = jsonObject.getAsJsonPrimitive("number").getAsInt();
-
-                    if (buildNumber != WynntilsMod.getBuildNumber()) {
-                        TextComponent clickable = new TextComponent("here.");
-                        clickable.setStyle(clickable
-                                .getStyle()
-                                .withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/update"))
-                                .withUnderlined(true)
-                                .withBold(true));
-                        McUtils.sendMessageToClient(new TextComponent("[Wynntils/Artemis]: Build " + buildNumber
-                                        + " is the latest version, but you are using build "
-                                        + WynntilsMod.getBuildNumber() + ". Please consider updating by clicking ")
-                                .append(clickable)
-                                .append(
-                                        new TextComponent(
-                                                "\nPlease note that Artemis is in alpha, and newer builds might introduce bugs."))
-                                .withStyle(ChatFormatting.GREEN));
-                    }
-
-                    return true;
-                })
-                .build();
-
-        RequestHandler handler = new RequestHandler();
-
-        handler.addAndDispatch(versionFetch, true);
+        UpdateManager.getLatestBuild().whenCompleteAsync((buildNumber, throwable) -> {
+            if (buildNumber != WynntilsMod.getBuildNumber()) {
+                TextComponent clickable = new TextComponent("here.");
+                clickable.setStyle(clickable
+                        .getStyle()
+                        .withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/update"))
+                        .withUnderlined(true)
+                        .withBold(true));
+                McUtils.sendMessageToClient(new TextComponent("[Wynntils/Artemis]: Build " + buildNumber
```

Comment:
Is it fine to use this method when its done async?

Reply:
You worry the player might leave the server?

Reply:
https://github.com/Wynntils/Artemis/pull/613/commits/de1ed08d61db8828cc993ddcb7d8b81762ec6bea